### PR TITLE
Set Authoritative Answer - DNSKEY (fix #210)

### LINKED ISSUE #210: Authoritative Answer
> eldinhadzic.com/DNSKEY (alg 14, id 36215): The Authoritative Answer (AA) flag was not set in the response. (163.172.141.143, UDP_0_EDNS0_32768_4096, UDP_0_EDNS0_32768_512)

http://dnsviz.net/d/eldinhadzic.com/V7A21A/dnssec/

**FIX???!!!** (working)

> middleware/dnssec/handler.go

```
// ServeDNS implements the middleware.Handler interface.
func (d Dnssec) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
// Intercept queries for DNSKEY, but only if one of the zones matches the qname, otherwise we let
        // the query through.
        if qtype == dns.TypeDNSKEY {
                for _, z := range d.zones {
                        if qname == z {
                                resp := d.getDNSKEY(state, z, do)
                                resp.Authoritative = true
                                state.SizeAndDo(resp)
                                w.WriteMsg(resp)
                                return dns.RcodeSuccess, nil
                        }
                }
        }
```

**Attention**

I don´t know if my "Fix" is suitable

### PR DESCRIPTION
**Pull request by elcore**
Hello @miekg 

:tada: :tada:  :tada: 

FIX + TEST

### CODE REVIEW

File: middleware/dnssec/handler_test.go [hunk 1]
```diff
@@ -129,6 +129,10 @@ func TestLookupDNSKEY(t *testing.T) {
 		}
 		resp := rec.Msg()
 
+		if resp.Authoritative != true {
```
Comment (miekg) [inline middleware/dnssec/handler_test.go]: Some small nits:
You don't need the `!= true`, just do a `if !resp.Authoritative`

File: middleware/dnssec/handler_test.go [hunk 2]
```diff
@@ -129,6 +129,10 @@ func TestLookupDNSKEY(t *testing.T) {
...
+			t.Errorf("Authoritative Answer should be true, got %v\n", resp.Authoritative)
```
Comment (miekg) [inline middleware/dnssec/handler_test.go]: you can just say, `got false` in stead of %v printing that value.

**Comment by miekg** [comment]
Thanks, look good.

Tests will prolly fail (fixing that now)

**Comment by elcore** [comment]
Do you mean Travis? Thank you for the style improvement.

**Comment by miekg** [comment]
Well, please fix those style issues :-)

**Comment by elcore** [comment]
I love it, go Go :smile:

Thank you @miekg

**Pull request merged**